config.toml.example

```diff
@@ -434,6 +434,11 @@ changelog-seen = 2
 # set this value to `true`.
 #debug-logging = rust.debug-assertions (boolean)
 
+# Whether or not to build rustc, tools and the libraries with randomized type layout
+#
+# Defaults to rust.debug value
+#randomize-layout = rust.debug (boolean)
```

Comment:
having this on by default for debugging seems kind of concerning, won't this make debugging non-deterministic?

hmm, I guess it only affects the initial build. but it seems kind of concerning to test this out on contributors when it's likely not the thing they're investigating.

Reply:
Well, depends on how one looks at those flags. If it's about enabling extra things that catch errors then it fits in with debug assets and overflow checks. If you actually want to attach a debugger then maybe not.

If you prefer I can make disable that inheritance and explicitly enable it in `src/ci` instead.

Reply:
Hmm, fair enough. `debug = true` already doesn't mean "a debuggable environment" because we don't enable debuginfo for local variables. So I'm ok with keeping it like you have it.

Reply:
It is deterministic. The seed is derived from the def path hash of the type: https://github.com/rust-lang/rust/blob/3fdd578d72a24d4efc2fe2ad18eec3b6ba72271e/compiler/rustc_middle/src/ty/mod.rs#L1883